saleor/webhook/payloads.py

```diff
@@ -1080,3 +1150,176 @@ def generate_excluded_shipping_methods_for_checkout_payload(
         ],
     }
     return json.dumps(payload, cls=CustomJsonEncoder)
+
+
+def _generate_order_prices_data_with_taxes(
+    order: "Order",
+    manager: PluginsManager,
+    lines: Optional[Iterable[OrderLine]] = None,
+) -> Dict[str, Decimal]:
+    shipping = order_calculations.order_shipping(order, manager, lines)
+    shipping_tax_rate = order_calculations.order_shipping_tax_rate(
+        order, manager, lines
+    )
+    total = order_calculations.order_total(order, manager, lines)
+    undiscounted_total = order_calculations.order_undiscounted_total(
+        order, manager, lines
+    )
+
+    return {
+        "shipping_price_net_amount": shipping.net.amount,
+        "shipping_price_gross_amount": shipping.gross.amount,
+        "shipping_tax_rate": shipping_tax_rate,
+        "total_net_amount": total.net.amount,
+        "total_gross_amount": total.gross.amount,
+        "undiscounted_total_net_amount": undiscounted_total.net.amount,
+        "undiscounted_total_gross_amount": undiscounted_total.gross.amount,
+    }
+
+
+def _generate_order_prices_data_without_taxes(
+    order: "Order",
+    use_gross_as_base_price: bool,
+) -> Dict[str, Decimal]:
+    def untaxed_price_amount(price: TaxedMoney) -> Decimal:
+        return quantize_price(
+            get_base_price(price, use_gross_as_base_price), order.currency
+        )
+
+    return {
+        "shipping_price_base_amount": untaxed_price_amount(order.shipping_price),
+        "total_base_amount": untaxed_price_amount(order.total),
+        "undiscounted_total_base_amount": untaxed_price_amount(
+            order.undiscounted_total
+        ),
+    }
+
+
+def generate_order_payload(
+    order: "Order",
+    requestor: Optional["RequestorOrLazyObject"] = None,
+    with_meta: bool = True,
+):
+    manager = get_plugins_manager()
+    lines = OrderLine.objects.prefetch_related("variant__product__product_type")
+
+    return _generate_order_payload(
+        order,
+        requestor,
+        with_meta,
+        order_prices_data=_generate_order_prices_data_with_taxes(order, manager, lines),
+        order_lines_payload=_generate_order_lines_payload_with_taxes(
+            order, manager, lines
+        ),
+        included_taxes_in_prices=include_taxes_in_prices(),
+    )
+
+
+def generate_order_payload_without_taxes(
+    order: "Order",
+    requestor: Optional["RequestorOrLazyObject"] = None,
+    with_meta: bool = True,
+):
+    lines = OrderLine.objects.prefetch_related("variant__product__product_type")
```

Comment:
I think that, You can do here select_related. This should reduce amount of queries